target/linux/ramips/dts/mt7621_dna_valokuitu-plus-ex400.dts

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	compatible = "dna,valokuitu-plus-ex400", "mediatek,mt7621-soc";
+	model = "DNA Valokuitu Plus EX400";
+
+	aliases {
+		led-boot = &led_status_red;
```

Comment:
```
                ethernet0 = &gmac0;
```

That will make U-Boot pass down the MAC address to Linux and then you don't need to read it via NVMEM/U-Boot env interface. Don't get me wrong, I appreciate the someone is testing NVMEM-on-UBI and the uboot-env-layout driver. However, I don't think it is needed here.

Reply:
With that aliasing I can get the mac? Was it there all along? Just asking because the nvmem layuts came up in the previous review round.

Are there some other aliases I should set? label-mac-device?

Reply:
> With that aliasing I can get the mac?

Yes, see https://github.com/u-boot/u-boot/blob/master/boot/fdt_support.c#L556

> Was it there all along?

Yes, U-Boot has been doing this since it supports device tree afaik.

> Are there some other aliases I should set? label-mac-device?

Yes, `label-mac-device` makes sense, and maybe also `ethernet1` (if there is `eth1addr` in the env -- otherwise your approach to use NVMEM for the wan MAC address is probably still most suitable)

Reply:
> Yes, see https://github.com/u-boot/u-boot/blob/master/boot/fdt_support.c#L556
> Yes, U-Boot has been doing this since it supports device tree afaik.

Unfortunately I cannot get it to work. The vendor u-boot is based on v2016.09-rc2 and the feature seems to there. Can the FDT be readonly so that the mac-address cannot be overwritten?